frame/staking/src/lib.rs

```diff
@@ -3341,14 +2682,15 @@ impl<T: Config> Module<T> {
 				let mut total_power = 0;
 				let mut others = Vec::with_capacity(support.voters.len());
 				support.voters.into_iter().for_each(|(nominator, weight)| {
-					let origin_weight = Self::power_of(&nominator) as ExtendedBalance;
+					let origin_weight = Self::power_of(&nominator) as Balance;
```

Comment:
What's the difference between `ExtendedBalance` and `Balance`, just renaming.

`Balance` looks like being the type of Currency.

Reply:
Both of these 2 type are `u128`

We use it for calculate our vote weight.

Reply:
I suggest to continue use ExtendedBalance.

Even though they are both u128, but for the weight in `support.voters` it is using `npos_election::ExtendedBalance`, we should use the same from the point of code readability and type system.

Reply:
Actually they use `Balance` as `Vote` now.

https://sourcegraph.com/github.com/paritytech/substrate/-/blob/frame/staking/src/lib.rs#L2366

`to_currency` retuan a `Balance` type.

I'd like to invoke a new type `Vote = u128` here, to clarify the usage.

Reply:
Substrate use `VoteWeight`(u64) and `ExtendedBalance`(u128) to represent vote in the election.

Here is an convert from `VoteWeight` to `ExtendedBalance`

https://github.com/paritytech/substrate/blob/2e34d473e6ba81b454eb039544f2162975c0b30b/primitives/npos-elections/src/helpers.rs#L38

Substrate's staking is more simple because they can directly use Currency as Vote, but here in our case, we should use `Power` (combination of ratio power from ring and kton).

When converting `ExtendedBalance` back to `Power` (from u128 to u32), we should have a `to_power` just like substrate's `to_currency` to strictly avoid overflow.

Anyway, I don't think we should use `Balance` here,  as we are using different thing as vote compared to substrate staking.

Reply:
We don't need to convert vote back to currency.

We record all balance value in the exposure which use for slashing.

Actually our power is the vote here.

I think we just need to alias `power` as `vote` (a `u128` type power).

Reply:
> We don't need to convert vote back to currency.
> 
> We record all balance value in the exposure which use for slashing.
> 
> Actually our power is the vote here.
> 
> I think we just need to alias `power` as `vote` (a `u128` type `power`).
> 
> ```rust
> // a `u128` type `Power`
> type Vote = u128;
> ```

Not convert vote back to currency, need convert vote back to power.

The purpose of this type converter is pure for strict overflow check.

Reply:
Could you point out which line we convert the `u128` back to `u32`?

Let me check.

Reply:
https://github.com/darwinia-network/darwinia-common/blob/a00d1031acb941161d79d9d06611a3b27539b60f/frame/staking/src/lib.rs#L3379